pkg/xen-tools/patches-4.19.0/20-return_zero_close_if_special_file.patch

```diff
@@ -0,0 +1,69 @@
+diff --git a/tools/qemu-xen/fsdev/virtfs-proxy-helper.c b/tools/qemu-xen/fsdev/virtfs-proxy-helper.c
```

Comment:
@roja, please, provide a description and explanation on this patch. You can make a commit and use `git format-patch` to generate the patch file. This is removing a CVE fix, I think we don't have too much to do for now, but we need to think about it. Perhaps we won't allow any special files in the future. For now, I think it's ok, since it's just keeping the current behavior. FYI: @rouming , @shjala

Reply:
I think I've also mentioned that previously along with a simple ask to make a separate commit for that patch. Of course this patch should have clear description, why it removes/disables the security check. This will safe a lot of time on the next upgrade iteration, when we need to recall why we do such a strange change, and without any clear description this looks completely bogus.

Reply:
How come we didn't have this issue after applying the security fix? according to the CVE patch we shouldn't be vulnerable but I'm not quite sure since I don't have time to set it up and test it, but if we are vulnerable it can lead to container escape.

Reply:
@roja-zededa would you break the upgrade into multiple commits? specifically for patches, now it is difficult to revert changes incrementally if we face issues in the futures.

Reply:
> How come we didn't have this issue after applying the security fix? according to the CVE patch we shouldn't be vulnerable but I'm not quite sure since I don't have time to set it up and test it, but if we are vulnerable it can lead to container escape, and obviously we don't want that, is there a way to keep the patch and fix the issue?

Theoretically we can block access to device files only (as description of the CVE says), that should be enough to fix our container use case and keep things fixed from security perspective (right?!). 

I do not know the details of plan9 implementation in qemu, but if rootfs of the container cooked that way, that there are files which use links or block devices, then we can/modify host memory from a container. That's easy to check.

But I do not think lack of security check should block this PR, because we can't get rid of plan9 easily, and proper fix can come later.

Anyway this does not exclude commits separation and clear description. We are still ignored here, which gets ridiculous to be frank

Reply:
> Theoretically we can block access to device files only (as description of the CVE says), that should be enough to fix our container use case and keep things fixed from security perspective (right?!).

What about the named pipes and sockets? it can lead to nasty things specifically with sockets. 

> But I do not think lack of security check should block this PR, because we can't get rid of plan9 easily, and proper fix can come later.

sure, I will keep it in my TODO list.

Reply:
> > How come we didn't have this issue after applying the security fix? according to the CVE patch we shouldn't be vulnerable but I'm not quite sure since I don't have time to set it up and test it, but if we are vulnerable it can lead to container escape, and obviously we don't want that, is there a way to keep the patch and fix the issue?
> 
> Theoretically we can block access to device files only (as description of the CVE says), that should be enough to fix our container use case and keep things fixed from security perspective (right?!).
> 
> I do not know the details of plan9 implementation in qemu, but if rootfs of the container cooked that way, that there are files which use links or block devices, then we can/modify host memory from a container. That's easy to check.
> 
> But I do not think lack of security check should block this PR, because we can't get rid of plan9 easily, and proper fix can come later.
> 
> Anyway this does not exclude commits separation and clear description. We are still ignored here, which gets ridiculous to be frank

@rouming @shjala I apologize, I was being side-tracked by the test failures. Will be splitting the PR into mutiple commits with detailed description ASAP.

Reply:
Thanks. No need to do asap, just don't forget, otherwise it will be too complicated in the future bisect, debug, understand.

Reply:
I'm OK with not including the security patch for now.